checkov/terraform/parser_functions.py

```diff
@@ -167,3 +168,28 @@ def _check_map_type_consistency(value: Dict) -> Dict:
     if had_string and had_something_else:
         value = {k: to_string(v) for k, v in value.items()}
     return value
+
+
+def handle_dynamic_values(conf: Dict[str, List[Any]]) -> None:
+    # recursively search for blocks that are dynamic
+    for block_name in conf.keys():
+        if isinstance(conf[block_name], dict):
+            handle_dynamic_values(conf[block_name])
+
+        # if the configuration is a block element, search down again.
+        if isinstance(conf[block_name], list) and len(conf[block_name]) > 0 and isinstance(conf[block_name][0], dict):
```

Comment:
```suggestion
        if isinstance(conf[block_name], list) and conf[block_name] and isinstance(conf[block_name][0], dict):
```
no need to calculate the length, if you just want to know, it is not empty. this way is also faster 🚀